pkg/koordlet/metricsadvisor/collectors/noderesource/node_resource_collector.go

```diff
@@ -126,12 +126,12 @@ func (n *nodeResourceCollector) collectNodeResUsed() {
 	}
 	nodeMetrics = append(nodeMetrics, cpuUsageMetrics)
 
-	for _, deviceCollector := range n.deviceCollectors {
+	for deviceType, deviceCollector := range n.deviceCollectors {
 		if metric, _ := deviceCollector.GetNodeMetric(); metric != nil {
 			nodeMetrics = append(nodeMetrics, metric...)
 		}
 		if info := deviceCollector.Infos(); info != nil {
-			n.metricDB.Set(koordletutil.GPUDeviceType, info)
+			n.metricDB.Set(deviceType, info)
```

Comment:
Nice catch, but perhaps `info.Type()` here.
The registered key in the `n.deviceCollectors` is exactly the collector name, though the GPU collector plugin's name is the same as its Type.